src/lib/y2partitioner/actions/controllers/fstabs.rb

```diff
@@ -0,0 +1,184 @@
+        #
+        # @see #missing_devices_error
+        #
+        # @return [Array<String>]
+        def selected_fstab_errors
+          [missing_devices_error].compact
+        end
+
+        # Imports mount points from the selected fstab
+        #
+        # Before importing, the current devicegraph is reset to the system one.
+        def import_mount_points
+          reset_current_graph
+          selected_fstab.filesystem_entries.each { |e| import_mount_point(e) }
+        end
+
+      private
+
+        # System devicegraph
+        #
+        # @return [Y2Storage::Devicegraph]
+        def system_graph
+          DeviceGraphs.instance.system
+        end
+
+        # Current devicegraph
+        #
+        # @return [Y2Storage::Devicegraph]
+        def current_graph
+          DeviceGraphs.instance.current
+        end
+
+        # Disk analyzer for the system devicegraph
+        #
+        # @return [Y2Storage::DiskAnalyzer]
+        def disk_analyzer
+          DeviceGraphs.instance.disk_analyzer
+        end
+
+        # Error when some devices in the selected fstab are missing in the current devicegraph
+        #
+        # @return [String, nil] nil if all devices are found
+        def missing_devices_error
+          return nil unless missing_devices?
+
+          _("Some required devices cannot be found in the system.")
+        end
+
+        # Whether any device in the selected fstab is missing in the system devicegraph
+        #
+        # @return [Boolean]
+        def missing_devices?
+          selected_fstab.filesystem_entries.any? { |e| e.device(system_graph).nil? }
+        end
+
+        # Initializes current devicegraph with system
+        def reset_current_graph
+          DeviceGraphs.instance.current = system_graph.dup
+        end
+
+        # Imports the mount point of a fstab entry
+        #
+        # The device in the fstab entry (first field) is formatted using the fileystem type
+        # indicated in the entry. In case the device is not a block device (e.g., NFS), the
+        # device is not formatted and only the mount point and mount options are assigned.
+        #
+        # @param entry [Y2Storage::SimpleEtcFstabEntry]
+        def import_mount_point(entry)
+          device = entry.device(current_graph)
+          return unless device
+
+          filesystem =
+            if device.is?(:blk_device)
+              format_device(device, entry.fs_type)
+            else
+              device
+            end
+
+          filesystem.mount_path = entry.mount_point
+          filesystem.mount_point.mount_options = entry.mount_options if entry.mount_options.any?
+        end
+
+        # Formats the device indicated in the fstab entry
+        #
+        # @param device [Y2Storage::BlkDevice]
+        # @param fs_type [Y2Storage::Filesystems::Type]
+        #
+        # @return [Y2Storage::Filesystems::Base]
+        def format_device(device, fs_type)
+          device.delete_filesystem
```

Comment:
NP: Shouldn't this be `remove_descendants`? Just trying to cover weird use cases like using a partition that nowadays belongs to an LVM or it's encrypted, but that it used to be used as a plain partition in the imported fstab.

I admit that's a very weird scenario, but with `delete_filesystem` there is a chance we crash on those scenarios, while `#remove_descendants` should be more robust.

Reply:
As discussed in another channels. Maybe a better fix is not to use `remove_descendants` (which is rather aggressive and can cause collateral damages to other mount points we want to import) but to ensure the code is only executed in cases in which the current `delete_filesystem` cannot fail and crash. That is, being more strict on deciding whether a mount point is importable or not.